plugins/lombok/src/main/java/de/plushnikov/intellij/plugin/util/PsiClassUtil.java

```diff
@@ -40,7 +41,9 @@
    */
   @NotNull
   public static Collection<PsiField> collectClassFieldsIntern(@NotNull PsiClass psiClass) {
-    if (psiClass instanceof PsiExtensibleClass) {
+    if (psiClass.isRecord()) {
+      return createRecordFields(psiClass);
```

Comment:
static fields of a record class are ignored. It doesn't matter for builder but influence e.g., the name conflicts here `de.plushnikov.intellij.plugin.processor.handler.FieldNameConstantsHandler#createFields`. Moreover, calling this method I would never assume that static fields are already filtered.
I also hesitate if it's ok to created additional fields which would be different to ones create by `com.intellij.psi.impl.RecordAugmentProvider` and which are already cached in different places but looks like that in lombok doesn't compare, so this one is probably worth mentioning in javadoc

Reply:
Lombok completely circumvents the augments here to prevent a recursion from happening with it's own augment.

Reply:
sorry, I don't understand your comment. I want to say that you need to include static fields ("ownFields") of a record class as well as generated record components.

Reply:
My comment was in reply to this:

> I also hesitate if it's ok to created additional fields which would be different to ones create by com.intellij.psi.impl.RecordAugmentProvider and which are already cached in different places but looks like that in lombok doesn't compare, so this one is probably worth mentioning in javadoc

This doesn't actually *create* any fields, merely collects them for the plugin to create it's own augments with.

Reply:
this need to be proved: if plugin propagate those fields anywhere or try to compare them with the `aClass.getFields()`  - IDE would be in trouble. Thus @mplushnikov suggests to patch just `BuilderHandler.java` as it's clear there that nothing bad can happen. Could you please do that or should we proceed ourselves? Thanks

Reply:
I've moved the logic for builder support to BuilderHandler and BuilderInfo. I've also moved the record field collection logic to AbstractFieldProcessor.